deps/spidershim/src/v8arraybuffer.cc

```diff
@@ -59,4 +66,76 @@ ArrayBuffer::GetContents()
   contents.byte_length_ = length;
   return contents;
 }
+
+Local<ArrayBuffer>
+ArrayBufferView::Buffer()
+{
+  Isolate* isolate = GetIsolate();
+  JSContext* cx = JSContextFromIsolate(isolate);
+  JS::RootedObject view(cx, &reinterpret_cast<JS::Value*>(this)->toObject());
+  bool shared;
+  JSObject* buf = JS_GetArrayBufferViewBuffer(cx, view, &shared);
+  if (!buf) {
+    return Local<ArrayBuffer>();
+  }
+
+  JS::Value bufVal;
+  bufVal.setObject(*buf);
+  return internal::Local<ArrayBuffer>::New(isolate, bufVal);
+}
+
+size_t
+ArrayBufferView::ByteOffset()
+{
+  JSObject* view = &reinterpret_cast<JS::Value*>(this)->toObject();
+  if (JS_IsTypedArrayObject(view)) {
+    return JS_GetTypedArrayByteOffset(view);
+  }
+
+  return JS_GetDataViewByteOffset(view);
+}
+
+size_t
+ArrayBufferView::ByteLength()
+{
+  JSObject* view = &reinterpret_cast<JS::Value*>(this)->toObject();
+  if (JS_IsTypedArrayObject(view)) {
+    return JS_GetTypedArrayByteLength(view);
+  }
+
+  return JS_GetDataViewByteLength(view);
+}
+
+#define TYPED_ARRAY_METHODS(TYPE) \
+  Local<TYPE> \
+  TYPE::New(Handle<ArrayBuffer> buffer, size_t offset, size_t length) \
+  { \
+    Isolate* isolate = buffer->GetIsolate(); \
+    JSContext* cx = JSContextFromIsolate(isolate); \
+    JS::RootedObject buf(cx, &reinterpret_cast<JS::Value*>(*buffer)->toObject()); \
+    JSObject* array = JS_New ## TYPE ## WithBuffer(cx, buf, offset, length); \
+    if (!array) { \
+      return Local<TYPE>(); \
+    } \
+    JS::Value arrayVal; \
+    arrayVal.setObject(*array); \
+    return internal::Local<TYPE>::New(isolate, arrayVal); \
+  } \
+\
+  TYPE* \
+  TYPE::Cast(Value* v) \
+  { \
+    assert(v->Is ## TYPE()); \
+    return static_cast<TYPE*>(v); \
+  }
+
+TYPED_ARRAY_METHODS(Uint8Array)
+TYPED_ARRAY_METHODS(Uint8ClampedArray)
+TYPED_ARRAY_METHODS(Int8Array)
+TYPED_ARRAY_METHODS(Uint16Array)
+TYPED_ARRAY_METHODS(Int16Array)
+TYPED_ARRAY_METHODS(Uint32Array)
+TYPED_ARRAY_METHODS(Int32Array)
+TYPED_ARRAY_METHODS(Float32Array)
+TYPED_ARRAY_METHODS(Float64Array)
```

Comment:
We already have `TYPED_ARRAY` macros in `valuemap.inc` which are intended to enumerate the different types of typed arrays.  We use them in `value.cc`.  It seems like you can reuse those macros by appending `Array` to `TYPE` occurrences in the macro defined above.  Can you please reuse the `valuemap.inc` macros here too?

Also, can you please #undef the macro after you've included valuemap.inc?